packages/ui/src/components/va-tree-view/VaTreeView.demo.vue

```diff
@@ -1,59 +1,352 @@
 <template>
   <VbDemo>
-    <VbCard>
-      <va-button @click="$refs.treeView.expand()">
-        EXPAND ALL
-      </va-button>
-      <va-button @click="$refs.treeView.collapse()">
-        COLLAPSE ALL
-      </va-button>
-
-      <va-tree-view ref="treeView">
-        <va-tree-category
-          label="Category text"
-          is-open
-        >
-          <va-tree-node>
-            Node text
-          </va-tree-node>
-          <va-tree-category
-            label="Category text"
-            is-open
-          >
-            <va-tree-node>
-              Node text
-            </va-tree-node>
-          </va-tree-category>
-        </va-tree-category>
+    <VbCard title="Default">
+      <va-tree-view
+        :nodes="nodes"
+        node-key="id"
+      />
+    </VbCard>
+    <VbCard title="Icons">
+      <va-tree-view
+        :nodes="nodesWithIcons"
+        node-key="id"
+        expand-all
+      >
+        <template #icon="{icon}">
+          <va-icon
+            v-if="icon"
+            :name="icon"
+            size="20px"
+          />
+        </template>
+      </va-tree-view>
+    </VbCard>
+    <VbCard title="Selection and icons">
+      <va-radio
+        v-for="(option, index) in selectionTypeOptions"
+        :key="index"
+        v-model="selectionType"
+        :option="option"
+      />
+      <br />
+      <p>
+        {{ JSON.stringify(selectedNodes) }}
+      </p>
+      <br />
+      <va-tree-view
+        v-model="selectedNodes"
+        :nodes="nodesWithIcons"
+        node-key="id"
+        expand-all
+        selectable
+        :selection-type="selectionType"
+      >
+        <template #icon="{icon}">
+          <va-icon
+            v-if="icon"
+            :name="icon"
+            size="20px"
+          />
+        </template>
       </va-tree-view>
     </VbCard>
-    <VbCard title="Color">
+    <VbCard title="With custom body">
       <va-tree-view
-        color="info"
+        :nodes="customBodyNodes"
+        node-key="id"
       >
-        <va-tree-category
-          label="Category text"
-          is-open
-        >
-          <va-tree-node>
-            Node text
-          </va-tree-node>
-          <va-tree-node>
-            Node text
-          </va-tree-node>
-        </va-tree-category>
+        <template #content="prop">
+          <div class="d-flex align--center">
+            <va-avatar
```

Comment:
Not a big deal, but probably no much sense to do something so complicated for demo, just outlining the slot and scoped props would do.